firebase-installations/src/main/java/com/google/firebase/installations/FirebaseInstallations.java

```diff
@@ -235,31 +235,41 @@ private void persistFid(String fid) throws FirebaseInstallationsException {
   }
 
   /**
-   * Registers the FID with FIS servers if FID is in UNREGISTERED state.
+   * Registers the FID with FIS servers if FID is in UNREGISTERED state. Also, refreshes auth token
+   * if expired.
    *
    * <p>Updates FID registration status to PENDING to avoid multiple network calls to FIS Servers.
    */
   private Task<String> registerFidIfNecessary(
       PersistedFidEntry persistedFidEntry, AwaitListener listener) {
     String fid = persistedFidEntry.getFirebaseInstallationId();
 
-    // Check if the fid is unregistered
-    if (persistedFidEntry.getRegistrationStatus() == RegistrationStatus.UNREGISTERED) {
-      updatePersistedFidWithPendingStatus(fid);
-      executeFidRegistration(persistedFidEntry, listener);
-    } else {
-      updateAwaitListenerIfRegisteredFid(persistedFidEntry, listener);
+    if (persistedFidEntry.getRegistrationStatus() != RegistrationStatus.UNREGISTERED
+        && !isAuthTokenExpired(persistedFidEntry)) {
+      return updateAwaitListenerIfRegisteredFid(persistedFidEntry, listener);
     }
 
+    updatePersistedFidWithPendingStatus(persistedFidEntry);
+    executeFISCalls(persistedFidEntry, listener);
     return Tasks.forResult(fid);
   }
 
-  private void updateAwaitListenerIfRegisteredFid(
+  private void executeFISCalls(PersistedFidEntry persistedFidEntry, AwaitListener listener) {
```

Comment:
Through the method name, it confusing that what the difference is between executeFISCalls and executeFidRegistration